src/asyncValidation.js

```diff
@@ -21,49 +23,46 @@ const callValidator = (validator, values, path, options, errors) => {
         return err;
       }
     });
-}
+};
 
 // Mix JOI validation with our own custom validators
 const asyncValidation = (joiSchema, customSchema) => {
   const validationFunction = async (values, options) => {
     const schema = Joi.object().keys(joiSchema);
     options.context.values = values;
 
-    return Joi.validate(values, schema, options, (errors, values) => {
-      if (errors && options.abortEarly) {
-        return errors;
-      } else if (! errors) {
-        errors = new Error();
-        errors.details = [];
-      }
+    const validated = Joi.validate(values, schema, options);
 
-      const promises = Object.keys(customSchema).reduce((accumulator, path) => {
-        if (! values[path]) {
-          return accumulator;
-        }
+    if (validated.error) {
+      return validated;
+    }
 
-        if (Array.isArray(customSchema[path])) {
-          customSchema[path].forEach(validator => {
-            accumulator.push(callValidator(validator, values, path, options, errors))
-          });
-        } else {
-          accumulator.push(callValidator(customSchema[path], values, path, options, errors));
-        }
+    const errors = new Error('ValidationError');
+    errors.details = [];
+    const promises = Object.keys(customSchema).reduce((accumulator, path) => {
+      if (!values[path]) {
         return accumulator;
-      }, []);
+      }
 
-      return Promise.all(promises)
-        .then(() => {
-          if (errors.details.length) {
-            return errors;
-          } else {
-            return values;
-          }
-        })
-        .catch((err) => {
-          return err;
+      if (Array.isArray(customSchema[path])) {
```

Comment:
Again, modifying this function shouldn't have been in the scope of this issue. This looks like code changes to make asyncValidation work with the changes you made to the validation object, all of which is outside the scope of this issue